### PR TITLE
add diskWaterMark to volume and server

### PR DESCRIPTION
**Pull request by bingoohuang**
1. diskWaterMark with disk total and free to master
2. master will check readonly and disk free > diskWaterMark when assigned fid
3. add util.LogFatalIfError and util.LogFatalIf to simplified coding
4. fix some typo
5. some refactoring to make code more readable

### CODE REVIEW

**Comment by chrislusf** [comment]
I personally do not like "3. add util.LogFatalIfError and util.LogFatalIf to simplified coding". It is some extra brain work for me to understand the logic.

File: weed/server/master_server.go [hunk 1]
```diff
@@ -50,11 +50,11 @@ func NewMasterServer(r *mux.Router, port int, metaFolder string,
 
 	var preallocateSize int64
 	if preallocate {
-		preallocateSize = int64(volumeSizeLimitMB) * (1 << 20)
+		preallocateSize = int64(volumeSizeLimit) * (1 << 20)
```
Comment (chrislusf) [inline weed/server/master_server.go:53]: This is wrong.

Comment (bingoohuang) [reply weed/server/master_server.go:53]: yes, there is no need to  * (1 << 20) again.

Comment (bingoohuang) [reply weed/server/master_server.go:53]: but the preallocateSize is not implemented yet, right?

**Comment by chrislusf** [comment]
1. no big diffs. Please split into separated diffs.
2. some changes are not desired.

**Comment by bingoohuang** [comment]
which part would you like to merge? I will pr the part which you will accept.

File: weed/server/master_server.go (hunk 1 above)
Comment (bingoohuang) [reply weed/server/master_server.go:53]: which part would you like to merge? I will pr the part which you will accept.

**Pull request closed without merging**